### PR TITLE
Updated Capsule Implementation

### PR DESCRIPTION
**Pull request by ns3284**
### Summary
Modified past version of the "cifar10_cnn_capsule.py" so that it works with the latest version. TF/Keras changed the batch_dot implementation, so this version skips batch_dot in favor for an included "caps_batch_dot" function.

### Related Issues
Relies on tf.matmul, because K.dot modifies the dimensions, due to the Numpy implementation.

### PR Overview

- [ n ] This PR requires new unit tests [y/n] (make sure tests are included)
- [ y ] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [ y ] This PR is backwards compatible [y/n]
- [ n ] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

**Comment by ns3284** [comment]
Fluke failure? shuffle of [0,1,2] == [0,1,2]. Did not fail this test on the previous pull request.

**Comment by ns3284** [comment]
Needs update. Temporary closing pull request.

**Comment by ns3284** [comment]
Fixed for case of shared_weight = False. Required input_shape.

File: examples/cifar10_cnn_capsule.py [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+
+"""now we reshape it as (batch_size, input_num_capsule, input_dim_capsule)
+then connect a Capsule layer.
+
+the output of final model is the lengths of 10 Capsule, whose dim=16.
+
+the length of Capsule is the proba,
+so thepochse problem becomes a 10 two-classification problem.
+"""
+x = Reshape((-1, 128))(x)
```
Comment (chaitanya11) [inline examples/cifar10_cnn_capsule.py]: @ns3284 Squash function has to be applied according to paper.

**Pull request closed without merging**